### ISSUE TITLE
editor+project crash if use_bvh=false

### ISSUE THREAD

**Issue by pseidemann** [godotengine/godot#70322]
### Godot version

v3.5.1.stable.official (6fed1ffa313c6760fa88b368ae580378daaef0f0)

### System information

macOS 13.1 (22C65)

### Issue description

when I enable the project setting `3d/godot_physics/use_bvh=false`, the editor will always crash when opening the project. also the project crashes on startup when running it via the command line

```
ERROR: Condition "_first != nullptr" is true.
   at: ~List (./core/self_list.h:108)
ERROR: Condition "_first != nullptr" is true.
   at: ~List (./core/self_list.h:108)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v3.5.1.stable.official (6fed1ffa313c6760fa88b368ae580378daaef0f0)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007ff804dd5c1d _sigtramp + 29
[2] 2   ???                                 0x00007ff7bf55c370 0x0 + 140702043718512
[3] GjkEpa2::MinkowskiDiff::get_support_without_margin(ShapeSW const*, Vector3 const&, float)
[4] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[5] DefaultAllocator::alloc(unsigned long)
[6] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[7] 7   dyld                                0x00007ff804a78310 start + 2432
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

* set `3d/godot_physics/use_bvh=false` in project settings
* open project in editor or start project

### Minimal reproduction project

I can't isolate what is causing this crash.
I'm using SurfaceTool, MeshDataTool and Navigation all at runtime

**Comment by lawnjelly** [comment]
It's a crash on this line in `Viewport::_process_picking(bool)`:
```
bool col = space->intersect_ray(from, from + dir * far, result, Set<RID>(), 0xFFFFFFFF, true, true, true);
```
```
1  typeinfo name for Object                           0x206c290 
2  Viewport::_process_picking     viewport.cpp   655  0x4e07ccb 
3  Viewport::_notification        viewport.cpp   399  0x4e05c73 
4  Viewport::_notificationv       viewport.h     91   0x4e284db 
5  Object::notification           object.cpp     927  0x600afab 
6  SceneTree::_notify_group_pause scene_tree.cpp 1133 0x4dd8439 
7  SceneTree::iteration           scene_tree.cpp 585  0x4dd7a4e 
8  Main::iteration                main.cpp       2309 0x2c5173f 
9  OS_X11::run                    os_x11.cpp     4035 0x2c05b1e 
10 main                           godot_x11.cpp  59   0x2bedc59 
```
Getting lots of errors from the octree:
```
ERROR: Condition "!E" is true.
   at: move (./core/math/octree_definition.inc:918)
```

**Comment by lawnjelly** [comment]
@rburing has bisected down to #55640 so it's due to me. :blush: 

It's probably we shifted some code around to octree that was no longer required for BVH. Am investigating. :+1:

**Comment by akien-mga** [comment]
Fixed by #70365.

**Comment by pseidemann** [comment]
thanks all!
